### DISCUSSION TITLE
[Feature Request] Ability to customize volume of "Upload Location" per site in Assets field

### DISCUSSION THREAD

**Discussion by benface** [craftcms/cms#8874] [Ideas]
### Description

Sometimes it's useful to [have a separate asset volume per site](https://github.com/craftcms/cms/issues/3586#issuecomment-664736495). However, it doesn't seem possible currently to force the user to use the proper asset volume when selecting assets in an Assets field, while adding or editing an entry in a specific site. It is possible to [customize the path *within* a volume to use the site's handle as a directory](https://craftcms.stackexchange.com/questions/30291/multi-site-setup-remote-asset-volume-to-store-assets-on-a-per-site-folder-basi), but I would really like to customize the volume itself because some users should only have access to some sites (and their volumes).

**Comment by J-Protz** [comment]
this says closed, but what is the solution?

**Reply by brandonkelly** [reply]
It’s not closed. There is a workaround though: create separate Assets fields for each site, each targeting a different volume, and show/hide the Assets fields within your field layout using a “Site” condition rule.

**Reply by JayBox325** [reply]
Are there any other solutions to this? As we're using Craft for a platform that could end up with 15/20 or (hopefully) more sites, this could become hard to maintain and scale.

We're investigating looking at whether different volumes per site or hopefully something with permissions can be done with the S3 plugin.

**Reply by brandonkelly** [reply]
Nothing comes to mind, sorry.
